lib/CXGN/BrAPI/v2/ObservationVariables.pm

```diff
@@ -109,27 +109,53 @@ sub search {
             push @and_wheres, "db.db_id = '$_'";
         }
     }
-    if (scalar(@dbxref_ids)>0){
-        my @db_names;
-        my @dbxref_accessions;
-        foreach (@dbxref_ids){
-            my ($db_name, $accession) = split ':', $_;
-            push @db_names, $db_name;
-            push @dbxref_accessions, $accession;
-        }
-        foreach (@db_names){
-            push @and_wheres, "db.name = '$_'";
-        }
-        foreach (@dbxref_accessions){
-            push @and_wheres, "dbxref.accession = '$_'";
+
+    # External reference id and reference source search
+    if (scalar(@dbxref_ids) > 0 || scalar(@dbxref_terms)>0) {
+        my $cv_id = $self->trait_ontology_cv_id;
+
+        my @sub_and_whereas;
```

Comment:
Is this variable name suppose to be `@sub_and_wheres`?

Reply:
Yep! Changed and pushed